File: apps/api/FinanceControl.Shared/Dtos/Response/UserPreferencesResponseDto.cs
namespace FinanceControl.Shared.Dtos.Response
{
    public class UserPreferencesResponseDto
    {
        public string CurrencyCode { get; set; } = string.Empty;
        public string Locale { get; set; } = string.Empty;
        public string? Country { get; set; }
        public string AnalyticsConfig { get; set; } = "[]";
    }
}

File: apps/mobile/lib/features/profile/data/dtos/user_preferences_response_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'user_preferences_response_dto.g.dart';

@JsonSerializable(createToJson: false)
class UserPreferencesResponseDto {
  const UserPreferencesResponseDto({
    required this.currencyCode,
    required this.locale,
    this.country,
  });

  final String currencyCode;
  final String locale;
  final String? country;

  factory UserPreferencesResponseDto.fromJson(Map<String, dynamic> json) =>
      _$UserPreferencesResponseDtoFromJson(json);
}
